### PR TITLE
Optimize scan of filter-only no-nulls integers

### PR DESCRIPTION
**Pull request by mbasmanova**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by yingsu00** [comment]
@mbasmanova Curious if you have the JMH benchmark numbers to show the gains?

**Comment by yingsu00** [comment]
@mbasmanova I restarted presto-hive tests which timed out. 
Thanks for making the change to add `isPositionalFilter()`. It makes my change easier too, and I will use it for my upcoming series of improvements. We can also optimize the no skipping and no filter case which I will include in my upcoming PR.

**Comment by mbasmanova** [comment]
> Curious if you have the JMH benchmark numbers to show the gains?

I don't have JVM benchmark for this case. I've been testing a specific workload on a cluster.

**Comment by mbasmanova** [comment]
@yingsu00 Thank you for review. I made the change you suggested.

`readOffset = offset + positionCount;`

File: presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java [hunk 1]
```diff
@@ -93,6 +93,9 @@
      */
     int getSucceedingPositionsToFail();
 
+    // Returns true if getPrecedingPositionsToFail or getSucceedingPositionsToFail may return non-zero values
```
Comment (highker) [inline presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java]: javadoc maybe

**Pull request merged**